internal/cli/tiltfile_result.go

```diff
@@ -0,0 +1,87 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	tiltanalytics "github.com/windmilleng/tilt/internal/analytics"
+	"github.com/windmilleng/tilt/internal/tiltfile"
+	"github.com/windmilleng/tilt/pkg/logger"
+	"github.com/windmilleng/tilt/pkg/model"
+)
+
+const TiltfileErrExitCode = 5
+
+type tiltfileResultDeps struct {
+	tfl tiltfile.TiltfileLoader
+}
+
+func newTiltfileResultDeps(tfl tiltfile.TiltfileLoader) tiltfileResultDeps {
+	return tiltfileResultDeps{
+		tfl: tfl,
+	}
+}
+
+func newTiltfileResultCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "tiltfile-result",
+		Short: "exec the tiltfile and print results as json",
+
+		Run: tiltfileResultPrintJSON,
+	}
+
+	cmd.Flags().StringVar(&fileName, "file", tiltfile.FileName, "Path to Tiltfile")
+
+	return cmd
+}
+
+func tiltfileResultPrintJSON(cmd *cobra.Command, args []string) {
+	l := logger.NewLogger(logger.DebugLvl, os.Stdout)
+	ctx := logger.WithLogger(context.Background(), l)
+	a, err := newAnalytics(l)
+	if err != nil {
+		failWithUnexpectedError(errors.Wrap(err, "Fatal error initializing analytics: %v"))
+	}
+
+	ctx = tiltanalytics.WithAnalytics(ctx, a)
+
+	deps, err := wireTiltfileResult(ctx, a)
+	if err != nil {
+		failWithUnexpectedError(errors.Wrap(err, "wiring dependencies"))
+	}
+
+	tlr := deps.tfl.Load(ctx, fileName, model.NewUserConfigState(args))
+	if tlr.Error != nil {
+		// Some errors won't JSONify properly--instead of messing with that, print the error
```

Comment:
I was mildly confused by this comment - it says instead of jsonifying the error, we're just going to print it, but then it jsonifies it instead of printing it?

How about just printing the error directly to stderr and exiting, and not doing any json?

Reply:
tiltfile output goes to stdout and json || error goes to stderr, assuming caller will use the exit code to figure out what kind of error (if any) it is. good point about the json, i'll just print the error i guess